pkg/datarepair/datarepair_test.go

```diff
@@ -130,31 +133,41 @@ func TestDataRepair(t *testing.T) {
 		satellite.Repair.Repairer.Loop.Pause()
 		satellite.Repair.Repairer.Limiter.Wait()
 
-		// kill nodes kept alive to ensure repair worked
+		// repaired segment should not contain any piece in the killed and DQ nodes
+		metainfo := satellite.Metainfo.Service
+		pointer, err = metainfo.Get(ctx, path)
+		require.NoError(t, err)
+
+		remotePieces = pointer.GetRemote().GetRemotePieces()
+		afterRepairNodes := make(map[storj.NodeID]struct{})
+		for _, piece := range remotePieces {
+			require.NotContains(t, nodesToKill, piece.NodeId, "there shouldn't be pieces in killed nodes")
+			require.NotContains(t, nodesToDisqualify, piece.NodeId, "there shouldn't be pieces in DQ nodes")
+
+			afterRepairNodes[piece.NodeId] = struct{}{}
+		}
+
+		nodesToKillForMinThreshold := len(remotePieces) - minThreshold
 		for _, node := range planet.StorageNodes {
-			if nodesToKeepAlive[node.ID()] {
+			if nodesToKillForMinThreshold == 0 {
+				break
+			}
+
+			if _, ok := afterRepairNodes[node.ID()]; ok {
```

Comment:
how do we guarantee that the nodes we are left with at the end of this loop are not just the nodes we kept alive before the repair? Consider the following (min threshold = 4):
Segment has nodes `a, b, c, d, e, f`. We kill/disqualify `e, f` so we repair off of `a, b, c, d` onto `g, h`. Our new repaired pointer is `a, b, c, d, g, h`.
I don't see what is stopping this section from just killing nodes `g, h`, and leaving us with the original ones we repaired from. I think the download should test that most if not all of the new nodes (`g, h`) are used for the download

Reply:
That's good point.

Reply:
I deleted that check from the original test without being aware, so I will add it back.